backend/src/main/java/com/woowacourse/gongcheck/auth/application/EntranceCodeProvider.java

```diff
@@ -0,0 +1,36 @@
+package com.woowacourse.gongcheck.auth.application;
+
+import com.woowacourse.gongcheck.exception.BusinessException;
+import org.springframework.stereotype.Component;
+
+@Component
+public class EntranceCodeProvider {
+
+    private final Hashable hashable;
+
+    public EntranceCodeProvider(final Hashable hashable) {
+        this.hashable = hashable;
+    }
+
+    public String createEntranceCode(final Long id) {
+        validateIdFormat(id);
+        return hashable.encode(String.valueOf(id));
+    }
+
+    public Long parseId(final String entranceCode) {
+        String result = hashable.decode(entranceCode);
+        try {
+            Long id = Long.parseLong(result);
+            validateIdFormat(id);
+            return id;
+        } catch (NumberFormatException | BusinessException e) {
+            throw new BusinessException("유효하지 않은 입장코드입니다.");
+        }
+    }
+
+    private void validateIdFormat(final Long id) {
+        if (id <= 0) {
```

Comment:
굳이인것 같긴하지만 ㅎㅎ; 이런 식으로도 표현하면 좋을 것 같다 라는 생각 ㅎ
```suggestion
        if (id < INITIAL_ID) {
```